lib/valkyrie/indexing/solr/indexing_adapter.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+require 'rsolr'
+
+module Valkyrie
+  module Indexing
+    module Solr
+      class IndexingAdapter
+        attr_reader :connection, :resource_indexer
+        # @param connection [RSolr::Client] The RSolr connection to index to.
+        # @param resource_indexer [Class, #to_solr] An indexer which is able to
+        #   receive a `resource` argument and then has an instance method `#to_solr`
+        def initialize(connection:, resource_indexer: Valkyrie::Persistence::Solr::MetadataAdapter::NullIndexer)
+          @connection = connection
+          @resource_indexer = resource_indexer
+        end
+
+        def conn
```

Comment:
Add a comment saying don't use this in new code.

consider deprecating the use of the rsolr connection directly to encourage going via search service methods instead.